aospy/test/test_calc_basic.py

```diff
@@ -39,109 +42,122 @@ def _test_files_and_attrs(calc, dtype_out):
     _test_output_attrs(calc, dtype_out)
 
 
-_BASIC_TEST_PARAMS = {
-    'proj': example_proj,
-    'model': example_model,
-    'run': example_run,
-    'var': condensation_rain,
-    'date_range': (datetime.datetime(4, 1, 1),
-                   datetime.datetime(6, 12, 31)),
-    'intvl_in': 'monthly',
-    'dtype_in_time': 'ts'
+_2D_DATE_RANGES = {
+    'datetime': (datetime.datetime(4, 1, 1), datetime.datetime(6, 12, 31)),
+    'datetime64': (np.datetime64('0004-01-01'), np.datetime64('0006-12-31')),
+    'cftime': (cftime.DatetimeNoLeap(4, 1, 1),
+               cftime.DatetimeNoLeap(6, 12, 31)),
+    'str': ('0004', '0006')
 }
+_3D_DATE_RANGES = {
+    'datetime': (datetime.datetime(6, 1, 1), datetime.datetime(6, 1, 31)),
+    'datetime64': (np.datetime64('0006-01-01'), np.datetime64('0006-01-31')),
+    'cftime': (cftime.DatetimeNoLeap(6, 1, 1),
+               cftime.DatetimeNoLeap(6, 1, 31))
+    # 'str': ('0006', '0006')  TODO: This should work once xarray version
+    # 0.10.5 is released
+}
+_2D_VARS = {'basic': condensation_rain, 'composite': precip}
+_2D_DTYPE_OUT_VERT = {'None': None}
+_2D_DTYPE_IN_VERT = {'None': None}
+_3D_VARS = {'3D': sphum}
+_3D_DTYPE_OUT_VERT = {'vert_int': 'vert_int'}
+_3D_DTYPE_IN_VERT = {'sigma': 'sigma'}
+_CASES = (
+    list(itertools.product(_2D_DATE_RANGES.items(), _2D_VARS.items(),
+                           _2D_DTYPE_IN_VERT.items(),
+                           _2D_DTYPE_OUT_VERT.items())) +
+    list(itertools.product(_3D_DATE_RANGES.items(), _3D_VARS.items(),
+                           _3D_DTYPE_IN_VERT.items(),
+                           _3D_DTYPE_OUT_VERT.items()))
+)
+_CALC_TESTS = {}
+for ((date_type, date_range), (test_type, var),
+     (vert_in_label, vert_in), (vert_out_label, vert_out)) in _CASES:
+    _CALC_TESTS['{}-{}-{}-{}'.format(
+        date_type, test_type, vert_in_label, vert_out_label)] = (
+            date_range, var, vert_in, vert_out)
+
+
+@pytest.fixture(params=_CALC_TESTS.values(), ids=list(_CALC_TESTS.keys()))
```

Comment:
Very cool usage of fixtures here!  And good in general to switch tests from unittest to pytest.